### PR TITLE
[libraqm] Add new port (0.6.0)

### PR DESCRIPTION
**Pull request by tobbi**

### CODE REVIEW

File: ports/libraqm/FindHarfbuzz.cmake [hunk 1]
```diff
@@ -0,0 +1,66 @@
+#.rst:
```
Comment (Neumann-A) [inline ports/libraqm/FindHarfbuzz.cmake]: Why is this required? harfbuzz installes a config which should be used.

Comment (tobbi) [reply ports/libraqm/FindHarfbuzz.cmake]: Should be fixed now.

**Comment by NancyLi1013** [comment]
Hi @tobbi, thanks for the new port. Here are the test results from the current CI system:

x64-windows|test|notes
|-----------|----|-----
|libraqm    |Fail|New Port

|x64-windows-static|test|notes
|------------------|----|-----
|libraqm           |Fail|New Port

|x64-osx|test|notes
|-------|----|-----
|libraqm|Skip|New Port

|arm64-windows|test|notes
|-------------|----|-----
|libraqm      |Skip|New Port

|x86-windows|test|notes
|-----------|----|-----
|libraqm    |Fail|New Port

|x64-linux|test|notes
|---------|----|-----
|libraqm  |Skip|New Port

|x64-uwp|test|notes
|-------|----|-----
|libraqm|Skip|New Port

|arm-uwp|test|notes
|-------|----|-----
|libraqm|Skip|New Port

[failureLogs.zip](https://github.com/microsoft/vcpkg/files/3230502/failureLogs.zip)

**Comment by tobbi** [comment]
Curious. Building worked just fine on my machine, which is a Win 10 x64. Does anyone have any idea what could have gone wrong?

**Comment by Neumann-A** [comment]
@tobbi: read the logs:
```
CMake Error at C:/vsts/_work/9/s/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "raqm" links to target "unofficial::glib::glib" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
```
you are simply missing a dependency

**Comment by tobbi** [comment]
> 
> 
> @tobbi: read the logs:
> 
> ```
> CMake Error at C:/vsts/_work/9/s/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
>   Target "raqm" links to target "unofficial::glib::glib" but the target was
>   not found.  Perhaps a find_package() call is missing for an IMPORTED
>   target, or an ALIAS target is missing?
> ```
> 
> you are simply missing a dependency

You see, when I built this locally, I didn't need that dependency. Here's a list of locally installed packages:
C:\vcpkg>vcpkg.exe list
bzip2:x64-windows                                  1.0.6-4          High-quality data compressor.
bzip2:x86-windows                                  1.0.6-4          High-quality data compressor.
freetype:x64-windows                               2.9.1-1          A library to render fonts.
freetype:x86-windows                               2.9.1-1          A library to render fonts.
fribidi:x64-windows                                58c6cb3          GNU FriBidi is an implementation of the Unicode ...
fribidi:x86-windows                                58c6cb3          GNU FriBidi is an implementation of the Unicode ...
harfbuzz:x64-windows                               2.4.0            HarfBuzz OpenType text shaping engine
harfbuzz:x86-windows                               2.4.0            HarfBuzz OpenType text shaping engine
harfbuzz[ucdn]:x64-windows                                          Builtin (UCDN) Unicode callbacks support
harfbuzz[ucdn]:x86-windows                                          Builtin (UCDN) Unicode callbacks support
libpng:x64-windows                                 1.6.37-1         libpng is a library implementing an interface fo...
libpng:x86-windows                                 1.6.37-1         libpng is a library implementing an interface fo...
libraqm:x64-windows                                0.5.0            A library for complex text layout
libraqm:x86-windows                                0.6.0            A library for complex text layout
ragel:x64-windows                                  6.10-1           Ragel State Machine Compiler
ragel:x86-windows                                  6.10-1           Ragel State Machine Compiler
zlib:x64-windows                                   1.2.11-5         A compression library
zlib:x86-windows                                   1.2.11-5         A compression library

So I'm a bit confused as to why glib is suddenly required.

**Comment by Neumann-A** [comment]
hmm on which commit is your checkout of vcpkg compared to current master of vcpkg? Could be some recent change which introduced the target in one of those dependent targets.

**Comment by tobbi** [comment]
I'm now on latest master. But even after rebasing, I can still build it fine. I rebuild every time with 
`vcpkg.exe remove libraqm && vcpkg.exe install libraqm`

**Comment by Neumann-A** [comment]
You have to rebuild `vcpkg upgrade --no-dry-run ` or start clean by deleting the folders installed, buildtress and packages.

**Comment by tobbi** [comment]
Even after running that command, the build still worked fine.

**Comment by Neumann-A** [comment]
hmm tested it locally and it also worked. Try to retrigger CI by either pushing a whitespace change or merging current master

**Comment by tobbi** [comment]
Done, let's see about the results.

**Comment by Neumann-A** [comment]
hmm seems like CI is again partly broken (at least the Preview CI is already finished an giving the same errors as before)

**Comment by tobbi** [comment]
Is the error message the same this time?

**Comment by vicroms** [comment]
CI keeps reporting failures on Windows, but I've been successful in installing it locally.
I'll take a look at what's causing the issues.

**Comment by vicroms** [comment]
The dependency to `unofficial::glib::glib` can come from `harfbuzz` when some features are enabled. However, that's not the case when `harfbuzz[ucdn]` is installed, so more investigation is required.

For `x64-windows-static` I did notice a problem with `fribidi`, the library produces a `libfribidi.a` binary on Windows and `libraqm` seems unable to find it because of it.

**Comment by vicroms** [comment]
As of now, it passes on most platforms, but fails on `x64-windows-static` due to issue an issue with`fribidi`'s meson build #6789.

I'm OK with merging this PR now and waiting for a new PR that addresses #6789 in the future.
